### PR TITLE
Fixed error for non-jpeg compressed images in original chunks

### LINKED ISSUE #6598: Segment Anything: Interaction Error Occurred: Request Failed with Status Code 500
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues) 
     - Found a similar one, but didn't quite helped me in resolving the one that I facing.
          -  (#6221)

### Steps to Reproduce (for bugs)

1. Install 2.5.2 cvat version in Ubuntu 22.04
2. Then, install the SAM using the serverless command mentioned in CVAT document 
3. Create a project, task and then upload few .tiff images in a task.

### Expected Behaviour

Using SAM to generate the mask/polygon.

### Current Behaviour

I have a job that consists of tiff images. For some of the images, the SAM interaction is working without any issue. However, on some, it's throwing the below error:

image

Then I checked the Grafana tool to trace the problem. Here is log:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/django/cvat/apps/lambda_manager/views.py", line 808, in func_wrapper
    data = func(*args, **kwargs)
  File "/home/django/cvat/apps/lambda_manager/views.py", line 899, in call
    return lambda_func.invoke(db_task, request.data, db_job=job, is_interactive=True, request=request)
  File "/home/django/cvat/apps/lambda_manager/views.py", line 302, in invoke
    "image": self._get_image(db_task, mandatory_arg("frame"), quality),
  File "/home/django/cvat/apps/lambda_manager/views.py", line 410, in _get_image
    image = frame_provider.get_frame(frame, quality=quality)
  File "/home/django/cvat/apps/engine/frame_provider.py", line 196, in get_frame
    chunk_reader = loader.load(chunk_number)
  File "/home/django/cvat/apps/engine/frame_provider.py", line 86, in load
    self.reader_class([self.get_chunk_path(chunk_id, self.quality, self.db_data)[0]]))
  File "/home/django/cvat/apps/engine/cache.py", line 57, in get_task_chunk_data_with_mime
    item = self._get_or_set_cache_item(
  File "/home/django/cvat/apps/engine/cache.py", line 49, in _get_or_set_cache_item
    item = create_function()
  File "/home/django/cvat/apps/engine/cache.py", line 59, in <lambda>
    create_function=lambda: self._prepare_task_chunk(db_data, quality, chunk_number),
  File "/home/django/cvat/apps/engine/cache.py", line 181, in _prepare_task_chunk
    writer.save_as_chunk(images, buff)
  File "/home/django/cvat/apps/engine/media_extractors.py", line 665, in save_as_chunk
    pil_image.save(output, format=pil_image.format if pil_image.format else self.IMAGE_EXT, quality=100, subsampling=0)
  File "/opt/venv/lib/python3.10/site-packages/PIL/Image.py", line 2353, in save
    save_handler(self, fp, filename)
  File "/opt/venv/lib/python3.10/site-packages/PIL/TiffImagePlugin.py", line 1770, in _save
    raise ValueError(
ValueError: quality setting only supported for 'jpeg' compression

### Your Environment

- Git hash commit (`git log -1`):
- Docker version `docker version` (e.g. Docker 17.0.05):
- Are you using Docker Swarm or Kubernetes?
- Operating System and version (e.g. Linux, Windows, MacOS):
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
    </details>

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #6598

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/engine/media_extractors.py [hunk 1]
```diff
@@ -662,7 +662,12 @@ def save_as_chunk(self, images, chunk_path):
                 output = io.BytesIO()
                 if self._dimension == DimensionType.DIM_2D:
                     pil_image = rotate_within_exif(Image.open(image))
-                    pil_image.save(output, format=pil_image.format if pil_image.format else self.IMAGE_EXT, quality=100, subsampling=0)
+                    # https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html
+                    if pil_image.format == 'TIFF':
+                        # use loseless lzw compression for tiff images
+                        pil_image.save(output, format='TIFF', compression='tiff_lzw')
+                    else:
+                        pil_image.save(output, format=pil_image.format if pil_image.format else self.IMAGE_EXT, quality=95, subsampling=0)
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:670]: Why quality 95?

File: cvat/apps/engine/media_extractors.py [hunk 2]
```diff
@@ -662,7 +662,12 @@ def save_as_chunk(self, images, chunk_path):
                 output = io.BytesIO()
                 if self._dimension == DimensionType.DIM_2D:
                     pil_image = rotate_within_exif(Image.open(image))
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:664]: I will suggest an optimization. Let's use the original file if it is without any rotations (output = content of the file).

Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:664]: A separate question is where the rotation should be handled. But leave it as is for now.

File: cvat/apps/engine/media_extractors.py (hunk 1 above)
Comment (bsekachev) [reply cvat/apps/engine/media_extractors.py:670]: Values above 95 should be avoided; 100 disables portions of the JPEG compression algorithm, and results in large fileswith hardly any gain in image quality.

https://jdhao.github.io/2019/07/20/pil_jpeg_image_quality/

In my test file size with quality 100 was twice more than with quality 95.

File: cvat/apps/engine/media_extractors.py (hunk 2 above)
Comment (bsekachev) [reply cvat/apps/engine/media_extractors.py:664]: Initially it was done to fix the bug with exif rotated images in original chunks which used in automatic annotations. 
As discussed, there was different options and it was decided to rotate original chunks.

Comment (bsekachev) [reply cvat/apps/engine/media_extractors.py:664]: But I agree, there may be optimization if rotate_within_exif did not rotate the file at all (no one if was triggered). In this case we may write original file.

File: cvat/apps/engine/media_extractors.py (hunk 1 above)
Comment (nmanovic) [reply cvat/apps/engine/media_extractors.py:672]: It is not the original file in the case. Can we try to save in PNG format in the case and use the original name?

Comment (nmanovic) [reply cvat/apps/engine/media_extractors.py:672]: There are some libraries which allow to rotate a jpeg file with 90 degrees step without loosing quality.

Comment (bsekachev) [reply cvat/apps/engine/media_extractors.py:672]: Okay, I will return 100 quality. 

> There are some libraries which allow to rotate a jpeg file with 90 degrees step without loosing quality.

This is not only about jpeg images, exif is about different formats.

File: cvat/apps/engine/media_extractors.py [hunk 3]
```diff
@@ -661,12 +661,23 @@ def save_as_chunk(self, images, chunk_path):
                 ext = os.path.splitext(path)[1].replace('.', '')
                 output = io.BytesIO()
                 if self._dimension == DimensionType.DIM_2D:
-                    pil_image = rotate_within_exif(Image.open(image))
-                    pil_image.save(output, format=pil_image.format if pil_image.format else self.IMAGE_EXT, quality=100, subsampling=0)
+                    pil_image, rotated = rotate_within_exif(Image.open(image))
```
Comment (nmanovic) [inline cvat/apps/engine/media_extractors.py:664]: I will recommend to have a separate method `is_image_rotated(img: Image)`. Or just open the image `img = Image.open(image)` and check rotation `img.getexif().get(ORIENTATION_EXIF_TAG) == ORIENTATION.NORMAL_HORIZONTAL. The code will be cleaner.

**Review by nmanovic — APPROVED** [review]
LGTM, let's wait linters and tests

**Pull request merged**